framework/src/st_things/things_stack/utils/things_network.c

```diff
@@ -318,13 +333,88 @@ void things_wifi_soft_ap_sta_left(void)
 	THINGS_LOG_V(TAG, "T%d --> %s", getpid(), __FUNCTION__);
 }
 
+void things_wifi_list_cleanup()
+{
+	if (g_wifi_scan_info != NULL) {
+		// free old wifi list
+
+		access_point_info_s* piter;
+		access_point_info_s* pdel;
+
+		piter = pdel = g_wifi_scan_info;
+
+		while (piter != NULL) {
+			pdel = piter;
+			piter = piter->next;
+
+			things_free(pdel);
+		}
+
+		g_wifi_scan_info = NULL;
+	}
+
+	g_wifi_count = 0;
+}
+
 void things_wifi_scan_done(wifi_manager_scan_info_s **scan_result, int res)
 {
 	THINGS_LOG_V(TAG, "T%d --> %s", getpid(), __FUNCTION__);
 	/* Make sure you copy the scan results onto a local data structure.
 	 * It will be deleted soon eventually as you exit this function.
 	 */
-//TO DO
+	if (scan_result == NULL) {
+		return;
+	}
+
+	things_wifi_list_cleanup();
+	wifi_manager_scan_info_s *wifi_scan_iter = *scan_result;
+	access_point_info_s *pinfo = NULL;
+	access_point_info_s *p_last_info = NULL;
+
+	while (wifi_scan_iter != NULL) {
+		if (wifi_scan_iter->ssid != NULL) {
+			pinfo = (access_point_info_s*)things_malloc(sizeof(access_point_info_s));
+			pinfo->next = NULL;
+			snprintf(pinfo->e_ssid, MAX_SSID_LEN, "%s", wifi_scan_iter->ssid);
+			snprintf(pinfo->bss_id, MAX_SSID_LEN, "%s", wifi_scan_iter->bssid);
+			snprintf(pinfo->signal_level, MAX_LEVEL_SIGNAL, "%d", wifi_scan_iter->rssi);
+			//strncpy(pinfo->security_key, wifi_scan_iter->ssid, MAX_SECUIRTYKEY_LEN);
+			//strncpy(pinfo->auth_type, wifi_scan_iter->ssid, MAX_TYPE_AUTH);
+			//strncpy(pinfo->enc_type, wifi_scan_iter->ssid, MAX_TYPE_ENC);
+
+			if (g_wifi_scan_info == NULL) {
+				g_wifi_scan_info = pinfo;
+			}
+			else {
```

Comment:
framework/src/st_things/things_stack/utils/things_network.c:388: ERROR: [BRC_M_SMT] else should follow close brace '}'

Reply:
@Hyunjun85 fixed.